runners/spark/src/main/java/org/apache/beam/runners/spark/stateful/SparkGroupAlsoByWindowViaWindowSet.java

```diff
@@ -338,6 +338,16 @@ public void outputWindowedValue(
               outputHolder.getWindowedValues();
 
           if (!outputs.isEmpty() || !stateInternals.getState().isEmpty()) {
+
+            Collection<TimerInternals.TimerData> expiredInternalTimers =
+                LateDataUtils.getExpiredTimers(
+                    timerInternals.getTimers(),
+                    windowingStrategy,
+                    timerInternals.currentInputWatermarkTime());
+
+            // Remove the expired timer from the timerInternals structure
+            expiredInternalTimers.forEach(timerInternals::deleteTimer);
```

Comment:
@bmv126 I would prefer that you add a method to LateDataUtils that drops expired timers rather than a method that just lists them.